src/DependencyInjection/ContainerFactory.php

```diff
@@ -323,6 +324,34 @@ private function validateParameters(array $parameters, array $parametersSchema):
 		if ($phpVersion['max'] < $phpVersion['min']) {
 			throw new InvalidPhpVersionException('Invalid PHP version range: phpVersion.max should be greater or equal to phpVersion.min.');
 		}
+
+		if (array_key_exists('featureToggles', $parameters) && is_array($parameters['featureToggles']) && (bool) $parameters['featureToggles']['bleedingEdge']) {
+			$this->validateExceptionClasses($parameters);
```

Comment:
You can put these conditions right next to:

```
		$container = $configurator->createContainer()->getByType(Container::class);
		$this->validateParameters($container->getParameters(), $projectConfig['parametersSchema']);
```

And pass ReflectionProvider properly from `$container` instead of obtaining it statically.

Reply:
Meaning do not call `validateExceptionClasses` from inside `validateParameters`, but instead from `create`.